Fix incorrect parsing of bill when 4-digit CFDA present in data

jakekreider opened the pull request:
### Ticket #193
Fixes #193 

## Description
From what I'm finding, CFDA's are strings with format NN.NNN. If the CFDA has a trailing zero, like `93.310`, the FFIS spreadsheet recognizes that field as a decimal and will store it as `93.31`. However the display format of the spreadsheet will include the trailing zero, masking the underlying issue. The way this handler works, it causes it to not see it as a CFDA (which it interprets as always marking the beginning of a data row) and instead treats it as a bill row. 

It seems like this is just a quirk of the spreadsheet we'll need to handle, rather than calling it a data issue.

This change modifies it to recognize CFDA's as `NN.NNN` OR `NN.NN`. Additionally, it adds a trailing zero as needed.

## Testing

1. tf up everything (e.g. `docker compose down; docker compose up -d; cd terraform; tflocal init;  tflocal apply -var-file=local.tfvars -auto-approve`)
2. Upload the [spreadsheet from the Slack thread](https://usdigitalresponse.slack.com/archives/C05GD4B55J9/p1689985019127809?thread_ts=1689984859.757369&cid=C05GD4B55J9), e.g. `awslocal s3 cp --sse AES256 download.xlsx s3://grants-ingest-grantssourcedata-000000000000-us-west-2/sources/2023/07/23/ffis/download.xlsx`
3. Wait a sec for the lambda to process
4. Get the file `awslocal s3 cp s3://grants-ingest-grantsprepareddata-000000000000-us-west-2/349/349292/ffis.org/v1.json 349292.json` and check contents.

This reproduces the issue before this code is applied, and should demonstrate the fix in this branch.

### Automated and Unit Tests
- [x] Added Unit tests

### Manual tests for Reviewer
- [x] Added steps to test feature/functionality manually

## Checklist
- [x] Provided ticket and description
- [x] Provided testing information
- [x] Provided adequate test coverage for all new code
- [x] Added PR reviewers

TylerHendrickson reviewed (commented):
Great find!

I'm now wondering what might happen for a (presumably valid?) CFDA value like `"02.980"`, where the first character is a zero and the spreadsheet could coerce the interpreted float value to a display string of `"2.98"`. If you think that's a reasonable enough (edge) case for us to anticipate, I left a suggestion that ought to do the trick. However, I'm also happy to accept the current solution if you think the work involved in updating how we detect the cell value as a CFDA would be unnecessary/ultimately more fragile.

jakekreider commented:
> Great find!
> 
> I'm now wondering what might happen for a (presumably valid?) CFDA value like `"02.980"`, where the first character is a zero and the spreadsheet could coerce the interpreted float value to a display string of `"2.98"`. If you think that's a reasonable enough (edge) case for us to anticipate, I left a suggestion that ought to do the trick. However, I'm also happy to accept the current solution if you think the work involved in updating how we detect the cell value as a CFDA would be unnecessary/ultimately more fragile.

Good call. The example spreadsheet won't add a leading zero so it seems less likely to cause confusion that way, but it could still happen if that's a valid CFDA. I added your suggestions (which make for cleaner number formatting anyway) and updated the test spreadsheet & test case to check for a leading zero as well.
